### ISSUE TITLE
Link color in view model

### ISSUE THREAD

**Issue by bogdan-pechounov** [Aghajari/AnnotatedText#4]
I am calling `fromHtml` inside the view model to keep track of a `currentWordIndex`, but I am not sure how to deal with the link color.

```
suspend fun String.toAnnotatedString() = withContext(Dispatchers.Default) {
    fromHtml(
        spanMappers = mapOf(
            URLSpan::class to {
//                this.linkColor = Color.None
                textDecoration = TextDecoration.None
            }
        )
    ).annotatedString
}
```

I  tried passing it via `savedStateHandle["linkColor"]` (done by compose navigation), but when the color scheme changes from the settings screen, everything refreshes and I go back to the main screen.

```
{
    val navController = rememberNavController()

    DataStoreProvider {
        AppTheme(
            darkTheme = Settings.darkTheme.value ?: return@DataStoreProvider,
            dynamicColor = Settings.dynamicColor.value ?: return@DataStoreProvider
        ) {
            val linkColor = MaterialTheme.colorScheme.tertiary // refreshes

            NavigationProvider(navigate = { navController.navigate(it) }) {
                NavHost(
                    navController = navController,
                    startDestination = Screen.Main.routeWithOptions
                ) {
                    composable(
                        Screen.Main.routeWithOptions,
                        arguments = listOf(
                            navArgument("linkColor") { defaultValue = linkColor.toArgb() },
                        )
                    )
                    {
                        MainScreen()
                    }

                    composable(Screen.Settings.route) {
                        SettingsScreen()
                    }
                }
            }
        }
    }
}
```

Is it possible to keep track of url spans and decide the color later (in a `@Composable`)?

**Comment by Aghajari** [comment]
Hello @bogdan-pechounov ,
I just published v1.02,

Now you can do this using `linkColorMapper: ((UrlSpan) -> Color?)?`

**Comment by bogdan-pechounov** [comment]
Awesome, thanks for your time!
